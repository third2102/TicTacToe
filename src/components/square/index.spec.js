import React from 'react';
import Square from './index.js';
import "./setupTest.js";
import { mount } from 'enzyme';

// sample unit tests
describe("square", ()=> {
    it("matches the snapshot", ()=>{
        const wrapper = mount(<Square value="test" isZardoz="ZARDOZ" />);
        expect(wrapper).toMatchSnapshot();
    })
    it("attaches classNames from value prop", () => {
        const wrapper = mount(<Square value="test" isZardoz=" " />);
        expect(wrapper.find('div').at(1).hasClass('test')).toBe(true);
    })
    it("attaches classNames from isZardoz", () => {
        const wrapper = mount(<Square value=" " isZardoz="zardoz" />);
        expect(wrapper.find('div').at(1).hasClass('zardoz')).toBe(true);
    })
    it("attaches classNames from isZardoz", () => {
        const mockupdate = jest.fn();
        const wrapper = mount(<Square onClick={mockupdate} value="" isZardoz="" />);
        wrapper.find('div').at(0).simulate('click');
        expect(mockupdate.mock.calls.length).toEqual(1);
    })
});