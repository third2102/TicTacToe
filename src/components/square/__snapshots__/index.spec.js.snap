// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`square matches the snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <square
    isZardoz="ZARDOZ"
    value="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "isZardoz": "ZARDOZ",
      "value": "test",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="square "
      >
        <div
          class="testZARDOZ"
        />
         
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="testZARDOZ"
          />,
          " ",
        ],
        "className": "square ",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="testZARDOZ"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "testZARDOZ",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "div",
        },
        " ",
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "isZardoz": "ZARDOZ",
        "value": "test",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="square "
        >
          <div
            class="testZARDOZ"
          />
           
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="testZARDOZ"
            />,
            " ",
          ],
          "className": "square ",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="testZARDOZ"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "testZARDOZ",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "div",
          },
          " ",
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
